sources/HUBViewControllerImplementation.m

```diff
@@ -241,16 +241,30 @@ - (void)didReceiveMemoryWarning
 
 #pragma mark - HUBViewController
 
-- (NSDictionary<NSNumber *, UIView *> *)visibleBodyComponentViews
+- (NSDictionary<NSIndexPath *, UIView *> *)visibleBodyComponentViewIndexPaths
 {
-    NSMutableDictionary<NSNumber *, UIView *> * const visibleViews = [NSMutableDictionary new];
-    
+    NSMutableDictionary<NSIndexPath *, UIView *> * const visibleViewIndexPaths = [NSMutableDictionary new];
+    NSMutableArray<HUBComponentWrapper *> *visibleComponents = [NSMutableArray array];
+
     for (HUBComponentCollectionViewCell * const cell in self.collectionView.visibleCells) {
         HUBComponentWrapper * const wrapper = [self componentWrapperFromCell:cell];
-        visibleViews[@(wrapper.model.index)] = HUBComponentLoadViewIfNeeded(wrapper);
+        [self addComponentWrapper:wrapper toArray:visibleComponents];
```

Comment:
Since this is acessed only from a single place, why does it need to be a method? (it could be made into a function in `HUBUtilities` if you think it warrants it).